execution.bs

```diff
@@ -6841,6 +6841,8 @@ enum class forward_progress_guarantee {
 
     2. <i>Synchronization:</i> This operation synchronizes with all `pop_front` operations on this object.
 
+    3. <i>Remarks:</i> If the thread of execution calling `run()` guarantees concurrent or parallel forward progress, then it guarantees parallel forward progress until `run()` returns; otherwise, it guarantees weakly parallel forward progress.
+
```

Comment:
From [this comment](https://github.com/NVIDIA/stdexec/pull/293#discussion_r761329669):

> _&lt;sigh&gt;_ I've gotten myself confused again. Since the execution of `run()` involves blocking for some condition to be satisfied by another thread (work is added to the queue, or another thread calls `finish()`), and since we can't guarantee that that will ever happen, it's possible that `run()` will never make progress. So doesn't that mean that the thread of execution calling `run()` has weakly parallel FP until it returns?